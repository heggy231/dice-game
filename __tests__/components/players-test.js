import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'; //eslint-disable-line
import Players from '../../js/components/players';
import Player from '../../js/components/player';

function setup() {
  const props = {
    players: [
      { id: 0, currentScore: 0, totalScore: 0 },
      { id: 1, currentScore: 0, totalScore: 0 }
    ],
    rounds: { player: 0 },
    winnerPlayer: jest.fn()
  };

  const enzymeWrapper = shallow(<Players {...props} />);

  return {
    props,
    enzymeWrapper
  };
}
   

describe('Players', () => {
    it('renders the div wrapper', () => {
        const { enzymeWrapper } = setup();
        expect(enzymeWrapper.find('div')).toHaveLength(1);
    });
    
    it('renders the the two players', () => {
        const { props } = setup();
        const tree = renderer.create(
          <Players {...props} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
