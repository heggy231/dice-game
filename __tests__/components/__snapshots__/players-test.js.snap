// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players renders the the two players 1`] = `
<div>
  <div
    className="player-0-panel active null"
  >
    <div
      className="player-name"
      id="name-0"
    >
      Player 
      1
    </div>
    <div
      className="player-score player-score-right"
      id="score-0"
    >
      0
    </div>
    <div
      className="player-current-box"
    >
      <div
        className="player-current-label"
      >
        Current
      </div>
      <div
        className="player-current-score"
        id="current-0"
      >
        0
      </div>
    </div>
  </div>
  <div
    className="player-1-panel null null"
  >
    <div
      className="player-name"
      id="name-1"
    >
      Player 
      2
    </div>
    <div
      className="player-score player-score-left"
      id="score-1"
    >
      0
    </div>
    <div
      className="player-current-box"
    >
      <div
        className="player-current-label"
      >
        Current
      </div>
      <div
        className="player-current-score"
        id="current-1"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
